src/composables/fungibleTokens.ts

```diff
@@ -90,7 +90,13 @@ export function useFungibleTokens() {
     const tokensFetchPromises = protocolsInUse.value.map(
       (protocol) => ProtocolAdapterFactory.getAdapter(protocol).fetchAvailableTokens(),
     );
-    const tokens: IToken[] = (await Promise.all(tokensFetchPromises)).flat();
+    // for each promise check if it returned null, if so, use cached data
+    // because it means that we couldn't fetch new data
+    const cachedTokens = tokensAvailable.value as ProtocolRecord<IToken[]>;
+
+    const tokens: IToken[] = (await Promise.all(tokensFetchPromises)).map(
+      (protocolTokens, index) => protocolTokens || cachedTokens[protocolsInUse.value[index]] || [],
```

Comment:
Maybe we don't need to create the `cachedTokens` and use `tokensAvailable.value` here?